### PR TITLE
Docker images for tags

### PR DESCRIPTION
**Pull request by whites11**
This PR changes the circle CI config to use the git tag name as docker image tag when building.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -17,11 +17,15 @@ jobs:
...
   version: 2
   build_and_update:
     jobs:
       - build:
           context:
             - architect
+          # Trigger the job also on git tag.
+          filters:
+            tags:
+              only: /^v[0-9]+\.[0-9]+\.[0-9]+(?:-[0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*)?(?:\+[0-9A-Za-z-]+)?$/
```
Comment (whites11) [inline .circleci/config.yml:31]: docker build fails if the docker tag is not a semver.
This is a regexp to validate semver strings

**Review by calvix — APPROVED** [review]
lgtmiwfy

**Review by fiunchinho — APPROVED** [review]
LGTM

File: .circleci/config.yml (hunk 1 above)
Comment (ubergesundheit) [reply .circleci/config.yml:31]: uh nice! Can you maybe document this in the architect-orb repository somewhere? Somewhere here? https://github.com/giantswarm/architect-orb/tree/master/docs

**Review by MarcelMue — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (kopiczko) [inline .circleci/config.yml:31]: Why not `/.*/`?

Comment (ubergesundheit) [reply .circleci/config.yml:31]: container image tags cannot contain `/` for example. (I'm sure there are others like spaces or `:`)

Comment (whites11) [reply .circleci/config.yml:31]: because the build of the container fails if the tag name is not a valid semver string

Comment (kopiczko) [reply .circleci/config.yml:31]: I see so it's about tag characters not allowed by the docker registry.

**Pull request merged**